libs/components/action-bars/src/lib/modules/summary-action-bar/summary-action-bar.component.ts

```diff
@@ -64,7 +64,16 @@ export class SkySummaryActionBarComponent implements AfterViewInit, OnDestroy {
   private idled = new Subject<boolean>();
 
   @ContentChild(SkySummaryActionBarSummaryComponent, { read: ElementRef })
-  private summaryElement: ElementRef;
+  private set summaryELement(element: ElementRef) {
+    this._summaryElement = element;
+    this.changeDetector.markForCheck();
+  }
+
+  private get summaryElement(): ElementRef {
+    return this._summaryElement;
+  }
+
+  private _summaryElement: ElementRef;
```

Comment:
How does this fix the issue at hand? From what I can see, it seems all you've done is set up getters and setters for this ElementRef.

Reply:
The key here is the `markForCheck`. When the `ngIf` is toggled externally the internal change detection doesn’t fire because the component is `OnPush`. Adding the `markForCheck` allows the internal change detection to run and for the template to update accordingly.